Kernel/FileSystem/FUSE/FUSEConnection.h

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2024, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <Kernel/FileSystem/FUSE/Definitions.h>
+#include <Kernel/FileSystem/OpenFileDescription.h>
+#include <Kernel/Library/KBuffer.h>
+
+namespace Kernel {
+
+class FUSEConnection : public RefCounted<FUSEConnection> {
+public:
+    static ErrorOr<NonnullRefPtr<FUSEConnection>> try_create(NonnullRefPtr<OpenFileDescription> description)
+    {
+        if (!description->is_device())
+            return Error::from_errno(EINVAL);
+
+        if (description->device()->class_name() != "FUSEDevice"sv)
+            return Error::from_errno(EINVAL);
+
+        auto* device = bit_cast<FUSEDevice*>(description->device());
+        TRY(device->initialize_instance(*description));
+
+        return TRY(adopt_nonnull_ref_or_enomem(new (nothrow) FUSEConnection(description)));
```

Comment:
What happens if we fail here? do we have a dangling instance within the `FUSEDevice`?